com.unity.netcode.gameobjects/Runtime/NetworkVariable/NetworkVariable.cs

```diff
@@ -96,6 +102,33 @@ public virtual T Value
             }
         }
 
+        // Compares two values of the same unmanaged type bitwise
+        // Ignoring any overriden value checks
+        // Size is fixed
+        private static unsafe bool ValueEquals(in T a, in T b)
```

Comment:
I wonder if we already have an `UnsafeUtility.Compare` or something similar that compares `unmanaged` types without `Object.Equals` or `EqualityComparer.Equals` to avoid boxing/GC. maybe some DOTS or Netcode for Entities people would know? I believe instead of reinventing it, hopefully we could use something that already exists (DRY).

Reply:
Yeah I tried to find something suitable but was unable to. But perhaps it's not all bad that we have control over that compare. In this case it's bitwise which lines up with our needs of the serializer.

Reply:
You could use [`UnsafeUtility.AddressOf`](https://docs.unity3d.com/ScriptReference/Unity.Collections.LowLevel.Unsafe.UnsafeUtility.AddressOf.html) + [`UnsafeUtility.MemCmp`](https://docs.unity3d.com/ScriptReference/Unity.Collections.LowLevel.Unsafe.UnsafeUtility.MemCmp.html) combination.

Reply:
if anything, `NetworkVariable::ValueEquals` is a reinvention of `UnsafeUtility.MemCmp`.
also, by using `UnsafeUtility.AddressOf`, you remove `unsafe` from the method and let `UnsafeUtility.AddressOf` to absorb unsafe things underneath.
both methods are available and maintained by Unity — can't justify reinventing the wheel here, please refactor.

Reply:
@TwoTenPvP I would argue that using UnsafeUtility is better in the end... because as with NativeArray etc... this is a unity supported solution, its not just well tested but very well tested and its something that has to be fast and always work since a lot of code relies on it. So in fact using it is the safest solution.

Reply:
Fair enough. I agree. Will refactor.

Reply:
> if anything, `NetworkVariable::ValueEquals` is a reinvention of `UnsafeUtility.MemCmp`. also, by using `UnsafeUtility.AddressOf`, you remove `unsafe` from the method and let `UnsafeUtility.AddressOf` to absorb unsafe things underneath. both methods are available and maintained by Unity — can't justify reinventing the wheel here, please refactor.

It still needs to be unsafe to deal with pointers though.